fitbenchmarking/controllers/horace_controller.py

```diff
@@ -86,10 +99,15 @@ def cleanup(self):
         else:
             self.flag = 0
 
-        self.final_params = self._fit_params['p'][0]
-
-        # Allow repeat calls to cleanup without falling over
-        try:
-            self.eng.evalc('horace_off')
-        except matlab.engine.MatlabExecutionError:
-            pass
+        self.final_params = np.array(self._fit_params['p'][0],
+                                     dtype=np.float64).flatten()
+
+        # # Allow repeat calls to cleanup without falling over
+        # try:
+        #     self.eng.evalc(
+        #         'if not(any(cellfun(@(x) x=="horace", persistent_vars)));'
+        #         ' horace_off;'
+        #         'end;'
+        #     )
+        # except matlab.engine.MatlabExecutionError:
+        #     pass
```

Comment:
Is it safe to uncomment this now?
We will need it if someone wants to run a single (non-horace) problem with all matlab solvers

Reply:
The reason why I commented this out is because when fitbenchmarking tries to recompute the jacobians at the end. It fails due to horace being turned off.  

Here is the error message:
![image](https://user-images.githubusercontent.com/83226114/219432319-95687643-b4de-4ef0-a600-c6177d828db1.png)

`hpc_config` is set when you turn horace_on().

For now I'll just add logic for it ignore those lines for spinw problems

Reply:
I thought that should be guarded by: `if not(any(cellfun(@(x) x=="horace", persistent_vars)));`
Maybe the persistent vars aren't working correctly then